### PR TITLE
Feature/account registration

### LINKED ISSUE #9: Register endpoint & bcrypt
**Is your feature request related to a problem? Please describe.**
Users need a way to create accounts

**Describe the solution you'd like**
- Endpoint: POST /api/auth/register
- Request: email: string, password: string
- Validate email format and password strength
- Check if email already exists -> 400 bad request
- Hash password with bcrypt (rounds = 10)
- Insert into users table
- Return success: true user: id, email

**Describe alternatives you've considered**
Could skip password strength validation, but it enhances user security and doesn't take a lot of work.

**Additional context**
Don't return password_hash in response. Handle duplicate email gracefully with a clear error message

### PR DESCRIPTION
**Pull request by mpolisch**
## Summary
Bootstraps the Express backend and implements POST /api/auth/register endpoint with Zod validation, bcrypt password hashing, and global error handling. Also established core structure that future BE PRs will follow.

Closes #9 

## Approach & Design Decisions
- Routes handle mapping, middleware handles validation, controllers handle request & response, service handles logic
- Zod validation middleware: validation.ts is reusable to accept any Zod schema. Validation happens before controller is ran, so bad requests never reach services
- Always return 201 regardless of whether email exists, using ON CONFLICT (email) DO NOTHING instead of separate select & insert to avoid race condition
- AppError extends error, adding statusCode and expose, which controls whether the message is sent to the client
- DB Pool configured with graceful shutdowns
- PublicUser type via Omit, meaning password_hash can never accidentally appear in response

## Security Considerations
- Passwords hashed with bcrypt (12 rounds). Plaintext never persisted or logged
- Email enumeration prevented - 201 response for new and duplicate emails
- Zod schema strips and validates input before services
- Password requirements: 8 chars minimum, one upper case, one lowercase, number, special character, no spaces allowed
- ON CONFLICT DO NOTHING - no race condition
- 5xx error messages not exposed to client
- DATABASE_URL validated at startp

## Testing
**What's covered:**
- [x] Unit tests
- [x] Edge cases (list below)

**Edge cases considered:**
- Duplicate email: service returns null and controller responds with 201
- DB connection failure: error propagates and caught by global error handler
- Password hashing called
- Missing / malformed body caught by Zod middleware

## Checklist
- [x] No secrets or credentials in code
- [x] Input validation in place
- [x] Error handling covers failure paths
- [x] Code is self-documenting or commented where non-obvious
- [x] No dead code or debug logs left in

## Notes
- Jest/ts-jest required a separate tsconfig.test.json with module: commonjs due to incompatibility between ts-jest and moduleResolution: NodeNext in TypeScript 6

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,14 +1,39 @@
...
+
 app.get('/', (req: Request, res: Response) => {
     res.send('Welcome to the backend server!');
 });
 
+app.use(errorHandler);
+
 const PORT: number = parseInt(process.env.PORT || '4000', 10);
 
 app.listen(PORT, () => {
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply src/index.ts:35]: out of scope for this project

File: src/routes/auth.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Router } from 'express';
+import * as authController from '../controllers/auth.js';
+import { validate } from '../middleware/validate.js';
+import { registerSchema } from '../utils/schemas.js';
+
+const router = Router();
+
+router.post('/register', validate(registerSchema), authController.register);
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply src/routes/auth.ts:8]: Verified the pipeline manually via Postman, not a valid concern

File: src/types/index.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+export interface User {
+    id: string;
+    email: string;
+    password_hash: string;
+    created_at: Date;
+    updated_at: Date;
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply src/types/index.ts:6]: Not a valid concern at this moment

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -29,3 +29,17 @@ jobs:
...
+        steps:
+            - uses: actions/checkout@v4
+            - uses: actions/setup-node@v4
+              with:
+                  node-version: 20.19.0
+                  cache: 'npm'
+            - name: Install dependencies
+              run: npm ci
+            - name: Run tests
+              run: npm test -- --passWithNoTests
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply .github/workflows/ci.yml]: updated.

File: src/services/auth.test.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import { registerUser } from './auth.js';
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply src/services/auth.test.ts:1]: resolved

File: tsconfig.test.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+    "extends": "./tsconfig.json",
+    "compilerOptions": {
+        "module": "commonjs",
+        "moduleResolution": "node",
+        "ignoreDeprecations": "6.0"
+        // CommonJS required for Jest/ts-jest compatibility
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply tsconfig.test.json]: resolved
